### PR TITLE
 Optionally parameterize slack notification header and job names

### PR DESCRIPTION
**Pull request by chestercharles**
#### Changes
- extracted all parameters read from github action context / inputs / etc into a 'getActionParams` function.
- added optional `header` input that will be the slack notification header
- added optional `name` input that will be the job name
- added test workflow for PRs

### CODE REVIEW

File: src/SlackMessage.js [hunk 1]
```diff
@@ -28,15 +28,15 @@ class SlackMessage extends SlackMessageRoot {
   constructor(token, message = { channel: '', ts: '' }) {
     super();
     if (!token) throw new Error('token is required');
-    this._token = token;
-    this._channel = message.channel;
-    this._ts = message.ts;
+    this.#_token = token;
+    this.#_channel = message.channel;
+    this.#_ts = message.ts;
```
Comment (chestercharles) [inline src/SlackMessage.js:33]: The only changes in this file are to use the hastag-prefixed instance variables. It looks like the original author of the class intended to use JS private class features but omitted the `#` inside the actual class.

Comment (chestercharles) [inline src/SlackMessage.js:33]: The only changes in this file are to use the hashtag-prefixed instance variables. It looks like the original author of the class intended to use JS private class features but omitted the `#` inside the actual class.

File: src/SlackMessage.mock.js [hunk 2]
```diff
@@ -8,30 +8,30 @@ class SlackMessage extends SlackMessageRoot {
 
   constructor(ts) {
     super();
-    this._mockTs = ts;
-    this._spies = {
+    this.#_mockTs = ts;
+    this.#_spies = {
```
Comment (chestercharles) [inline src/SlackMessage.mock.js:12]: The only changes in this file are to use the hashtag-prefixed instance variables. It looks like the original author of the class intended to use JS private class features but omitted the `#` inside the actual class.

**Review by areida — APPROVED** [review]

File: action.yml [hunk 3]
```diff
@@ -16,6 +16,11 @@ inputs:
     description: "Github token with actions:read permission"
   jobStatus:
     description: "Give me job.status please!"
+  name:
+    description: "Defaults to job name"
```
Comment (spruce-bruce) [inline action.yml:20]: If you don't provide a default, isn't this then a required option?

Comment (chestercharles) [reply action.yml:20]: I want it to default to the job name per the job context (like it did before) but this value cannot be specified in the yaml definition file.

**Pull request merged**